src/Sulu/Bundle/WebsiteBundle/Controller/DefaultController.php

```diff
@@ -76,13 +76,23 @@ protected function resolveRedirectUrl($redirectUrl, $requestUri)
         $redirectInfo = $this->parseUrl($redirectUrl);
         $requestInfo = $this->parseUrl($requestUri);
 
-        $url = sprintf('%s://%s', $requestInfo['scheme'], $redirectInfo['host']);
+        $url = sprintf('%s://%s', $requestInfo['scheme'], $requestInfo['host']);
+
+        if (isset($redirectInfo['host'])) {
+            $url = sprintf('%s://%s', $requestInfo['scheme'], $redirectInfo['host']);
+        }
 
         if (isset($requestInfo['port'])) {
             $url .= ':' . $requestInfo['port'];
         }
 
-        if (isset($redirectInfo['path'])) {
+        if (
+            isset($redirectInfo['path'])
+            && (
+                // if requested url not starting with redirectUrl it need to be added
+                !isset($requestInfo['path'])
+                || strpos($requestInfo['path'], $redirectInfo['path'] . '/') !== 0
+        )) {
```

Comment:
That looks really complicated, and probably still doesn't solve the issue I was talking about before... Is it valid that both, the request and redirect, have a path?